tests/deploy/test_scheduler_agent.py

```diff
@@ -107,7 +108,7 @@ async def execute(
         self.execute_count += 1
         result = (
             const.HandlerResourceState.failed
-            if resource_details.attributes.get(FAIL_DEPLOY, False) is True
+            if int(str(resource_details.attributes.get(FAIL_COUNT, 0))) >= self.execute_count
```

Comment:
Why the `str()` cast?

Reply:
Pycharm was complaining (and I thought mypy would complain to) about casting an object to int. It turns out mypy did not care